app/controllers/impersonate_controller.rb

```diff
@@ -104,40 +112,34 @@ def display_error_msg
     end
   rescue StandardError
     flash[:error] = @message
-    redirect_to :back
+    redirect_back
   end
 
   # Main operation
-  def do_main_operation(user)
-    if user
-      check_if_user_impersonateable
-    else
-      display_error_msg
-    end
+  def do_impersonate_operation(user)
+    check_if_user_impersonateable if user
+    display_error_msg
   end
 
   # Main operation, method used to break the functions in impersonate controller and bring out 2 functionalities at same level,
   # checking if user impersonateable, if not throw corresponding error message
+
   def impersonate
```

Comment:
This entire impersonate method is poorly written. `display_error_msg` method is terrible, and I believe should be removed entirely. This is how I think this logic should be implemented. You can check the implementation present in the master branch and take inspiration from there as that is working as expected.
* Implement another method that checks the input params and should show error if parameters are invalid (for example, existence of special characters or if field is empty. This method should be called before the impersonate is called. You can check out how `before_action` can be used for this case.
* After the input is checked, we can go on with our logic

Reply:
Resolved.
Created a method to check the validity of params . Using before_action ,this method is first called before the impersonate method. Removed the display_error_message method and its dependencies in the code